web/pgadmin/preferences/static/js/components/PreferencesComponent.jsx

```diff
@@ -533,7 +533,39 @@ export default function PreferencesComponent({ ...props }) {
     }
 
     if (_data.length > 0) {
-      save(_data, data);
+      // Check whether layout is changed from Workspace to Classic.
+      let raise_layout_warning = false;
+      for (const [key, value] of Object.entries(data.current)) {
```

Comment:
I think we can avoid looping by checking if layout is changed in `_data`. If yes then check for the value.